dbms/src/Server/Server.cpp

```diff
@@ -1325,6 +1326,27 @@ int Server::main(const std::vector<std::string> & /*args*/)
         settings.max_memory_usage_for_all_queries.getActualBytes(server_info.memory_info.capacity),
         settings.bytes_that_rss_larger_than_limit);
 
+    if (global_context->getSharedContextDisagg()->isDisaggregatedComputeMode())
+    {
+        // No need to have local index scheduler.
+    }
+    else if (global_context->getSharedContextDisagg()->isDisaggregatedStorageMode())
+    {
+        // There is no compute task in write node.
+        // Set the pool size to 80% of logical cores and 60% of memory
+        // to take full advantage of the resources and avoid blocking other tasks like writes and compactions.
+        global_context->initializeGlobalLocalIndexerScheduler(
+            server_info.cpu_info.logical_cores * 8 / 10,
```

Comment:
Add a protect `std::max(1, ...)` for the pool_size?

Reply:
done